torchvision/transforms/functional.py

```diff
@@ -908,10 +908,14 @@ def affine(
 
         return F_pil.affine(img, matrix=matrix, resample=resample, fillcolor=fillcolor)
 
-    # we need to rescale translate by image size / 2 as its values can be between -1 and 1
-    translate = [2.0 * t / s for s, t in zip(img_size, translate)]
+    if img_size[0] == img_size[1]:
+        # we need to rescale translate by image size / 2 as its values can be between -1 and 1
+        translate_f = [2.0 * t / s for s, t in zip(img_size, translate)]
+    else:
+        # if rectangular image, we should not rescale translation part
+        translate_f = [1.0 * t for t in translate]
 
-    matrix = _get_inverse_affine_matrix([0, 0], angle, translate, scale, shear)
+    matrix = _get_inverse_affine_matrix([0, 0], angle, translate_f, scale, shear)
```

Comment:
Here, I do not like that depending on if image is square or not matrice's translation part is normalized or not...

Reply:
Can't we just make everything follow the same code-path?

Reply:
The problem is that in case of square image we use `affine_grid` and it requires rescaled translation part, so we compute `matrix` here with rescaled `translate_f`. In case of rectangular images, we use custom affine grid implementation `_gen_affine_grid` where coords normalization is applied a posteriori and we need to deal with `matrix` where translation part is not normalized. Online normalization,denormalization of `matrix` is not evident neither. That's why there are two pathes.

Reply:
can't we just use `_gen_affine_grid` everywhere?